testsuite/forge/src/backend/k8s/mod.rs

```diff
@@ -21,6 +25,56 @@ impl K8sFactory {
             treasury_compliance_key,
         }
     }
+
+    // Instead of having the root_key and treasury_compliance_key explicitly, we should
+    // fetch these from the cluster internally via Vault, assuming the cluster environment
+    // has provided access
+    pub fn new_internal() -> std::result::Result<K8sFactory, Error> {
+        let vault_addr = env::var("VAULT_ADDR")
+            .map_err(|_| format_err!("Expected environment variable VAULT_ADDR"))?;
+        let vault_cacert = env::var("VAULT_CACERT")
+            .map_err(|_| format_err!("Expected environment variable VAULT_CACERT"))?;
+        let vault_token = env::var("VAULT_TOKEN")
+            .map_err(|_| format_err!("Expected environment variable VAULT_TOKEN"))?;
+
+        let vault_cacert_path = PathBuf::from(vault_cacert.clone());
+
+        let mut vault_cacert_file = File::open(vault_cacert_path)
+            .map_err(|_| format_err!("Failed to open VAULT_CACERT file at {}", &vault_cacert))?;
+        let mut vault_cacert_contents = String::new();
+        vault_cacert_file
+            .read_to_string(&mut vault_cacert_contents)
+            .map_err(|_| format_err!("Failed to read VAULT_CACERT file at {}", &vault_cacert))?;
+
+        let vault = VaultStorage::new(
+            vault_addr,
+            vault_token,
+            Some(vault_cacert_contents),
+            None,
+            false,
+            None,
+            None,
+        );
+        vault.available()?;
+        let root_key = base64::encode(
```

Comment:
Maybe we can just store the bytes directly instead of encoding them, cleaning up the other places where we're decoding.

Reply:
yes, let's modify ```load_root_key``` and ```load_tc_key``` under swamr.rs which just do ```Ed25519PrivateKey::try_from``` with bytes since we do not load encoded base64 string anymore